### PR TITLE
feat: Extend sns subscription data

### PR DESCRIPTION
**Pull request by irmatov**
Add hcl files

### CODE REVIEW

File: plugins/source/aws/resources/services/sns/subscriptions.hcl [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+output_directory = "."
+add_generate     = true
+
+description_modifier "remove_read_only" {
+  words = ["  This member is required."]
+}
+
+resource "aws" "sns" "subscriptions" {
+  path = "github.com/cloudquery/cloudquery/plugins/source/aws/resources/services/sns.Subscription"
+  ignoreError "IgnoreAccessDenied" {
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/sns/subscriptions.hcl:12]:
```suggestion
  ignoreError "IgnoreCommonErrors" {
```

Just to make the description match the actual value

File: plugins/source/aws/resources/services/sns/topics.hcl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+output_directory = "."
+add_generate     = true
+
+description_modifier "remove_read_only" {
+  words = ["  This member is required."]
+}
+
+resource "aws" "sns" "topics" {
+  path = "github.com/cloudquery/cloudquery/plugins/source/aws/resources/services/sns.Topic"
+  ignoreError "IgnoreAccessDenied" {
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/sns/topics.hcl:12]:
```suggestion
  ignoreError "IgnoreCommonErrors" {
```

File: plugins/source/aws/resources/services/sns/types.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	PendingConfirmation *bool
+	// True if raw message delivery is enabled for the subscription.
+	RawMessageDelivery *bool
+	// When specified, sends undeliverable messages to the specified Amazon SQS dead-letter queue.
+	RedrivePolicy *string
+	// The ARN of the IAM role that has permission to write to the Kinesis Data Firehose delivery stream and has Amazon SNS listed as a trusted entity.
+	SubscriptionRoleArn *string
+}
+
+// Amazon SNS topic.
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/sns/types.go:28]: Interesting! Why couldn't we use the SDK-provided `Topic` here?

**Review by hermanschaaf — APPROVED** [review]
LGTM :+1: Just left some minor nits and questions

**Comment by yevgenypats** [comment]
Can you please rebase?

File: plugins/source/aws/resources/services/sns/types.go (hunk 3 above)
Comment (irmatov) [reply plugins/source/aws/resources/services/sns/types.go:30]: SDK provides most of the data in the form of `map[string]string`, see https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/sns#GetTopicAttributesOutput

File: plugins/source/aws/resources/services/sns/topics.hcl (hunk 2 above)
Comment (irmatov) [reply plugins/source/aws/resources/services/sns/topics.hcl]: Updated, thanks

File: plugins/source/aws/resources/services/sns/subscriptions.hcl (hunk 1 above)
Comment (irmatov) [reply plugins/source/aws/resources/services/sns/subscriptions.hcl]: Updated, thanks

**Pull request merged**